src/parquet/column/properties.h

```diff
@@ -82,28 +82,41 @@ static constexpr int64_t DEFAULT_PAGE_SIZE = 1024 * 1024;
 static constexpr bool DEFAULT_IS_DICTIONARY_ENABLED = true;
 static constexpr int64_t DEFAULT_DICTIONARY_PAGE_SIZE_LIMIT = DEFAULT_PAGE_SIZE;
 static constexpr int64_t DEFAULT_WRITE_BATCH_SIZE = 1024;
+static constexpr bool DEFAULT_COLLECT_STATISTICS = false;
 static constexpr Encoding::type DEFAULT_ENCODING = Encoding::PLAIN;
 static constexpr ParquetVersion::type DEFAULT_WRITER_VERSION =
     ParquetVersion::PARQUET_1_0;
 static std::string DEFAULT_CREATED_BY = "Apache parquet-cpp";
 static constexpr Compression::type DEFAULT_COMPRESSION_TYPE = Compression::UNCOMPRESSED;
 
-using ColumnCodecs = std::unordered_map<std::string, Compression::type>;
+class PARQUET_EXPORT ColumnSettings {
```

Comment:
I'm ok with this class (although it should be named `ColumnProperties`) but not with the outcome at the end. Currently it is possible to select a custom encoding for a column but fallback to the file-global defaults for all other settings. With this new class, once we have set one property on a column-basis, for this column we have to either set all other file-global defaults explicitly on this column or we'll get the parquet-cpp defaults.